pdns/pdns_recursor.cc

```diff
@@ -1231,9 +1231,8 @@ static PolicyResult handlePolicyHit(const DNSFilterEngine::Policy& appliedPolicy
     return PolicyResult::NoAction;
 
   case DNSFilterEngine::PolicyKind::Custom:
-    ret.clear();
     res = RCode::NoError;
-    {
+    if (post && ret.size() == 0) { // can happen with NS matches, those do not fill the result
```

Comment:
I'm not sure we should treat that case differently?

Reply:
For the moment I decided to just copy what has been done until now for NS name or IP triggers. The current code always trows an exception if a NS policy is hit, and then does a resolve without RPZ processing starting with the custom record from the policy. I have trouble seeing how to modify the code in SyncRes  to handle the NS RPZ cases in SyncRes itself.

So I decided to keep the current behaviour (with an ugly hack, true). Need to revisit this, but I might need some help in that.